Mapbox3DTiles.js

```diff
@@ -1,492 +1,897 @@
+                if (this.debugColor) {
+                  gltf.scene.traverse(child => {
+                    if (child instanceof THREE.Mesh) {
+                      child.material.color = this.debugColor;
+                    }
+                  })
+                }
+                this.tileContent.add(gltf.scene);
+              }, (error) => {
+                throw new Error('error parsing gltf: ' + error);
+              }
+            );
+          } catch (error) {
+            console.error(error);
+          }
+          break;
+        case 'pnts':
+          try {
+            let pnts = new PNTS(url);
+            let pointData = await pnts.load();            
+            let geometry = new THREE.BufferGeometry();
+            geometry.setAttribute('position', new THREE.Float32BufferAttribute(pointData.points, 3));
+            let material = new THREE.PointsMaterial();
+            material.size = this.styleParams.pointsize != null ? this.styleParams.pointsize : 1.0;
+            if (this.styleParams.color) {
+              material.vertexColors = THREE.NoColors;
+              material.color = new THREE.Color(this.styleParams.color);
+              material.opacity = this.styleParams.opacity != null ? this.styleParams.opacity : 1.0;
+            } else if (pointData.rgba) {
+              geometry.setAttribute('color', new THREE.Float32BufferAttribute(pointData.rgba, 4));
+              material.vertexColors = THREE.VertexColors;
+            } else if (pointData.rgb) {
+              geometry.setAttribute('color', new THREE.Float32BufferAttribute(pointData.rgb, 3));
+              material.vertexColors = THREE.VertexColors;
+            }
+            this.tileContent.add(new THREE.Points( geometry, material ));
+            if (pointData.rtc_center) {
+              let c = pointData.rtc_center;
+              this.tileContent.applyMatrix4(new THREE.Matrix4().makeTranslation(c[0], c[1], c[2]));
+            }
+            this.tileContent.add(new THREE.Points( geometry, material ));
+          } catch (error) {
+            console.error(error);
+          }
+          break;
+        case 'i3dm':
+          throw new Error('i3dm tiles not yet implemented');
+          break;
+        case 'cmpt':
+          throw new Error('cmpt tiles not yet implemented');
+          break;
+        default:
+          throw new Error('invalid tile type: ' + type);
+      }
+    }
+  }
+  unload(includeChildren) {
+    this.unloadedTileContent = true;
+    this.totalContent.remove(this.tileContent);
+
+    //this.tileContent.visible = false;
+    if (includeChildren) {
+      this.unloadedChildContent = true;
+      this.totalContent.remove(this.childContent);
+      //this.childContent.visible = false;
+    } else  {
+      this.childContent.visible = true;
+    }
+    if (this.debugLine) {
+      this.totalContent.remove(this.debugLine);
+      this.unloadedDebugContent = true;
+    }
+    // TODO: should we also free up memory?
```

Comment:
Free memory to be picked up by garbage collector can be achieved by:
this.childContent = null;
this.tileContent = null;

Not doing this here, it would require reloading and reparsing the tiles when zooming in again. Maybe wait for 10? secs and then remove if data still unused? TODO